plugins/web-worker-offloading/third-party.php

```diff
@@ -0,0 +1,65 @@
+<?php
+/**
+ * Third-party integration loader for Web Worker Offloading.
+ *
+ * @since 0.1.0
+ * @package web-worker-offloading
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+/**
+ * Adds scripts to be offloaded to a worker.
+ *
+ * @since 0.1.0
+ * @access private
+ *
+ * @param string[] $script_handles Script handles.
+ */
+function wwo_mark_scripts_for_offloading( array $script_handles ): void {
+	add_filter(
+		'print_scripts_array',
+		static function ( $to_do ) use ( $script_handles ) {
+			$worker_script_handles = array_intersect( (array) $to_do, $script_handles );
+			foreach ( $worker_script_handles as $worker_script_handle ) {
+				wp_script_add_data( $worker_script_handle, 'worker', true );
+			}
+			return $to_do;
+		}
+	);
+}
+
+/**
+ * Loads third party plugin integrations for active plugins.
+ *
+ * @since 0.1.0
+ * @access private
+ */
+function wwo_load_third_party_integrations(): void {
+	$plugins_with_integrations = array(
+		// TODO: google-site-kit.
+		// TODO: seo-by-rank-math.
+		'woocommerce',
+	);
+
+	// Load corresponding file for each string in $plugins if the WordPress plugin is installed and active.
+	$active_plugin_slugs = array_filter(
+		array_map(
+			static function ( $plugin_file ) {
+				if ( is_string( $plugin_file ) && str_contains( $plugin_file, '/' ) ) {
```

Comment:
`str_starts_with` would be more robust. Otherwise this results in false positives for things like `not-woocommerce`.

Even more robust would be to check for a specific constant or function defined by the plugin, as the plugin might not necessarily be in that exact folder.

For example on my local dev site, Site Kit is installed in `plugins/site-kit-wp`, not `plugins/google-site-kit`

Reply:
> `str_starts_with` would be more robust. Otherwise this results in false positives for things like `not-woocommerce`.

I don't think this is right because the `str_contains()` is just checking whether it is a directory-based plugin and not a file-based plugin. The exact match of `woocommerce` is handled below with the `array_intersect()`.

> Even more robust would be to check for a specific constant or function defined by the plugin, as the plugin might not necessarily be in that exact folder.

That's true. I was wanting to avoid manually keeping track of those things, but I suppose it's not so bad.

Reply:
Done in 74b93c79